src/main/java/org/openstreetmap/atlas/checks/validation/linear/edges/RoundaboutConnectorCheck.java

```diff
@@ -0,0 +1,156 @@
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.geography.atlas.walker.OsmWayWalker;
+import org.openstreetmap.atlas.tags.HighwayTag;
+import org.openstreetmap.atlas.tags.JunctionTag;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.scalars.Angle;
+
+/**
+ * This check looks for roundabout connectors that intersect the roundabout at too sharp an angle.
+ * This is generally caused by one way roads drawn in the wrong direction or two way roads that
+ * should be one way.
+ *
+ * @author bbreithaupt
+ */
+public class RoundaboutConnectorCheck extends BaseCheck<Long>
+{
+    private static final long serialVersionUID = -5311314357995383430L;
+
+    private static final Double ONE_WAY_THRESHOLD_DEFAULT = 100.0;
+    private static final Double TWO_WAY_THRESHOLD_DEFAULT = 130.0;
+    private static final String ONE_WAY_INSTRUCTION = "This way, id:{0,number,#}, is connected to a roundabout at too sharp and angle. It may be digitized backwards";
+    private static final String TWO_WAY_INSTRUCTION = "This way, id:{0,number,#}, is connected to a roundabout at too sharp and angle to be a two way road.";
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays.asList(ONE_WAY_INSTRUCTION,
+            TWO_WAY_INSTRUCTION);
+
+    // Maximum angle for a turn in or out of a roundabout from a one way road.
+    private final Angle oneWayThreshold;
+    // Maximum angle for a turn in or out of a roundabout from a two way road.
+    private final Angle twoWayThreshold;
+
+    /**
+     * The default constructor that must be supplied. The Atlas Checks framework will generate the
+     * checks with this constructor, supplying a configuration that can be used to adjust any
+     * parameters that the check uses during operation.
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public RoundaboutConnectorCheck(final Configuration configuration)
+    {
+        super(configuration);
+        this.oneWayThreshold = this.configurationValue(configuration, "threshold.one-way",
+                ONE_WAY_THRESHOLD_DEFAULT, Angle::degrees);
+        this.twoWayThreshold = this.configurationValue(configuration, "threshold.two-way",
+                TWO_WAY_THRESHOLD_DEFAULT, Angle::degrees);
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Edge && !this.isFlagged(object.getOsmIdentifier())
+                && !JunctionTag.isRoundabout(object)
+                && ((Edge) object).highwayTag().isMoreImportantThan(HighwayTag.SERVICE);
```

Comment:
Is there a possibility for the minimum highway type to change? Should we make this a configurable?

Reply:
Yeah, I guess we may want to limit this at some point. I initially did it like this because I was thinking we would not want to go below service, but we may want to restrict this to higher highway values for more targeted flags. I will change this to be configurable.